Tests/TestUtils/TestNetworkService.swift

```diff
@@ -13,25 +13,53 @@
 @testable import AEPEdge
 @testable import AEPServices
 import Foundation
-
-/// Overriding NetworkService used for functional tests when extending the FunctionalTestBase
-class FunctionalTestNetworkService: NetworkService {
-    private var receivedNetworkRequests: [NetworkRequest: [NetworkRequest]] = [NetworkRequest: [NetworkRequest]]()
-    private var responseMatchers: [NetworkRequest: HttpConnection] = [NetworkRequest: HttpConnection]()
-    private var expectedNetworkRequests: [NetworkRequest: CountDownLatch] = [NetworkRequest: CountDownLatch]()
+import XCTest
+
+/// Overriding NetworkService used for functional tests when extending the TestBase
+class TestNetworkService: NetworkService {
+    private var usingMockNetworkRequestMode: Bool
+    private var receivedNetworkRequests: [NetworkRequest: [NetworkRequest]] = [:]
+    /// Matches outgoing `NetworkRequest`s with their corresponding `HttpConnection` response.
+    /// When using mocks, the `HttpConnection` response can be set using `setResponseConnectionFor(networkRequest:responseConnection:)`
+    private var responseMatchers: [NetworkRequest: HttpConnection] = [:]
+    private var expectedNetworkRequests: [NetworkRequest: CountDownLatch] = [:]
     private var delayedResponse: UInt32 = 0
+    
+    init(usingMockNetworkRequestMode: Bool) {
+        self.usingMockNetworkRequestMode = usingMockNetworkRequestMode
+        super.init()
+    }
 
     override func connectAsync(networkRequest: NetworkRequest, completionHandler: ((HttpConnection) -> Void)? = nil) {
-        FunctionalTestBase.log("Received connectAsync to URL \(networkRequest.url.absoluteString) and HTTPMethod \(networkRequest.httpMethod.toString())")
+        TestBase.log("Received connectAsync to URL \(networkRequest.url.absoluteString) and HTTPMethod \(networkRequest.httpMethod.toString())")
         if var requests = receivedNetworkRequests[networkRequest] {
             requests.append(networkRequest)
+            receivedNetworkRequests[networkRequest] = requests
         } else {
             receivedNetworkRequests[networkRequest] = [networkRequest]
         }
 
+        // Switch for mocked and real network requests
+        guard usingMockNetworkRequestMode else {
+            // Using real network requests and receiving real responses
+            super.connectAsync(networkRequest: networkRequest, completionHandler: { (connection: HttpConnection) in
+                print("Received connectAsync to URL \(networkRequest.url.absoluteString) and HTTPMethod \(networkRequest.httpMethod.toString())")
+                
+                let responseInserted = self.setResponseConnectionFor(networkRequest: networkRequest, responseConnection: connection)
+                if !responseInserted {
+                    XCTFail("Unable to insert response because one already exists for network request: \(networkRequest)")
+                }
```

Comment:
tests with multiple requests+responses sounds like a common use-case, we should allow this

Reply:
I was referring more to case no 3, where you call sendEvent x 5 times and expect 5 requests to /interact and 5 responses with 200 ok